### PR TITLE
`General`: Unify delete and reset button icons

### LINKED ISSUE #7520: `General`: Use trash can as icon for delete buttons
### Is your feature request related to a problem?

Artemis would be more intuitive to use if the delete button would not use an `x` but a trash can instead.

### Describe the solution you'd like

Exchange the `x` for all `Delete Buttons` with a trash can

@milljoniaer @rabeatwork @florian-glombik

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
closes #7520 
The Reset and Delete-buttons both had a "x" as icon (on most pages). This prevents us from using icon buttons (without text).

### Description
This PR sets the reset icon to an "undo-arrow" and delete icon to a trash bin.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with exercises

1. Log in to Artemis
2. Navigate to Course Administration
3. Navigate through Exercise Overview and Detail Pages
4. Verify that icons are correct set.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [competency-management.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/competencies/competency-management/competency-management.component.ts.html) | 87.7% | ✅ |
| [course-management-tab-bar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts.html) | 95.89% | ✅ |
| [course-update.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/course-update.component.ts.html) | 88.11% | ✅|
| [tutorial-group-free-period-row-buttons.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/tutorial-groups/tutorial-groups-management/tutorial-free-periods/tutorial-free-periods-management/tutorial-group-free-period-row-buttons/tutorial-group-free-period-row-buttons.component.ts.html) | 92.1% | ✅ |
| [tutorial-group-session-row-buttons.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/tutorial-groups/tutorial-groups-management/tutorial-group-sessions/tutorial-group-sessions-management/tutorial-group-session-row-buttons/tutorial-group-session-row-buttons.component.ts.html) | 87.75% | ✅ |
| [tutorial-group-row-buttons.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/tutorial-groups/tutorial-groups-management/tutorial-groups/tutorial-groups-management/tutorial-group-row-buttons/tutorial-group-row-buttons.component.ts.html) | 89.58% | ✅ |
| [exam-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exam/manage/exams/exam-detail.component.ts.html) | 93.44% | ✅ |
| [exercise-groups.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exam/manage/exercise-groups/exercise-groups.component.ts.html) | 89.42% | ✅ |
| [file-upload-exercise.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/file-upload/manage/file-upload-exercise.component.ts.html) | 93.1% | ✅ |
| [modeling-exercise.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/modeling/manage/modeling-exercise.component.ts.html) | 94.64% | ✅ |
| [programming-exercise-detail.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/programming-exercise-detail.component.ts.html) | 71.23% | ❌* |
| [programming-exercise.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/programming/manage/programming-exercise.component.ts.html) | 88.11% | ✅ |
| [quiz-exercise.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/quiz/manage/quiz-exercise.component.ts.html) | 89.47% | ✅ |
| [exam-exercise-row-buttons.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/exam-exercise-row-buttons/exam-exercise-row-buttons.component.ts.html) | 100% | ✅ |
| [non-programming-exercise-detail-common-actions.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/exercise-detail-common-actions/non-programming-exercise-detail-common-actions.component.ts.html) | 93.22% | ✅ |
| [participation-submission.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/participation-submission/participation-submission.component.ts.html) | 94.77% | ✅ |
| [participation.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/shared/participation/participation.component.ts.html) | 87.02% | ✅ |
| [text-exercise-row-buttons.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/manage/text-exercise/text-exercise-row-buttons.component.ts.html) | 95.45% | ✅ |
| [text-exercise.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/exercises/text/manage/text-exercise/text-exercise.component.ts.html) | 92.3% | ✅ |
| [lecture-attachments.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/lecture/lecture-attachments.component.ts.html) | 97.19% | ✅ |
| [lecture-unit-management.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/lecture/lecture-unit/lecture-unit-management/lecture-unit-management.component.ts.html) | 89.24% | ✅ |
| [lecture.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/lecture/lecture.component.ts.html) | 96.42% | ✅ |
| [delete-dialog.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5813/latest/artifact/shared/Coverage-Report-Client-Tests/app/shared/delete-dialog/delete-dialog.component.ts.html) | 92.1% | ✅ |

* The test coverage of the programming-exercise-detail file is fixed in #7632 . To avoid merge conflicts and double work, I didn't fix them within this PR

### Screenshots
image
image

### CODE REVIEW

File: src/main/webapp/app/exercises/shared/participation-submission/participation-submission.component.html [hunk 1]
```diff
@@ -29,7 +29,8 @@ <h4 *ngIf="exercise?.assessmentDueDate">
         [scrollbarH]="true"
     >
         <ngx-datatable-column name="Id" prop="id" [minWidth]="60" [width]="60">
-            <ng-template ngx-datatable-cell-template let-value="value">
+            <ng-template ngx-datatable-cell-template let-value="value"
+                >ex
```
[the remark this answers is not in the document]
Comment (milljoniaer) [reply src/main/webapp/app/exercises/shared/participation-submission/participation-submission.component.html]: yes, thanks, I removed it

**Review by egekurt123 — DISMISSED** [review]
Tested on TS5, works as expected

**Review by AntonGeTUM — DISMISSED** [review]
Tested on ts11, worked as expected

**Review by xHadie — CHANGES_REQUESTED** [review]
_Manually tested on TS6_

In the reset-confirmation modal the reset button still has an "x" as icon.
image

**Review by LenaKahle — APPROVED** [review]
Approve, I couldn't find any issues apart from the one that @xHadie found. Tested on ts11.

**Review by LenaKahle — APPROVED** [review]
Reapprove.

**Comment by Ceiridge** [comment]
Works as expected on ts2

image
image

**Review by xHadie — APPROVED** [review]

**Comment by MaximilianJG** [comment]
Tested on Ts3. All looks as expected to me! 

Screenshot 2023-12-06 at 13 41 54
Screenshot 2023-12-06 at 13 42 09

**Review by georgyia — APPROVED** [review]
tested manually on ts2, works as expected

File: src/main/webapp/app/shared/delete-dialog/delete-dialog.component.html [hunk 2]
```diff
@@ -44,7 +44,13 @@ <h4 class="modal-title" [ngSwitch]="actionType">
             [ngSwitch]="actionType"
         >
             <span *ngIf="submitDisabled"><fa-icon [icon]="faSpinner" [spin]="true">&nbsp;</fa-icon>&nbsp;</span>
-            <fa-icon [icon]="useFaCheckIcon ? faCheck : faTimes"></fa-icon>&nbsp;
+            <fa-icon *ngSwitchCase="actionTypes.Delete" [icon]="useFaCheckIcon ? faCheck : faTrash"></fa-icon>&nbsp;
+            <fa-icon *ngSwitchCase="actionTypes.Reset" [icon]="useFaCheckIcon ? faCheck : faUndo"></fa-icon>
+            <fa-icon *ngSwitchCase="actionTypes.Cleanup" [icon]="useFaCheckIcon ? faCheck : faTimes"></fa-icon>
```
Comment (florian-glombik) [inline src/main/webapp/app/shared/delete-dialog/delete-dialog.component.html:49]: I do not understand this change - why are further options added here?

Arent Delete, Rest and Cleanup displayed twice with the spans below?

**Review by florian-glombik — CHANGES_REQUESTED** [review]
I do not understand the change in the `delete-dialog` - could you explain that to me? 

Is there a reason why no component for  `<fa-icon [icon]="faTrash"></fa-icon>` was introduced? 
As this PR does still unify the icons and thereby improves Artemis this is fine for me, but I think a component just for the Icon would have been beneficial here _(= only one place where the icon needs to be adjusted)_

File: src/main/webapp/app/shared/delete-dialog/delete-dialog.component.html (hunk 2 above)
Comment (milljoniaer) [reply src/main/webapp/app/shared/delete-dialog/delete-dialog.component.html:49]: The spans below provide different text for the different action types, whereas this changes adds different icons for different actiontypes.

Comment (florian-glombik) [reply src/main/webapp/app/shared/delete-dialog/delete-dialog.component.html:49]: Ah I see, I think its a slightly better approach to have one switch case and combine the icon and text within one element (e.g. wrapped with `<></>`), but for this small change I do not have a strong opinion

**Pull request merged**